### DISCUSSION TITLE
Configure which GPU to use in a Multi-GPU-Setup

### DISCUSSION THREAD

**Discussion by chris-31337** [rustdesk/rustdesk#5073] [Feature Request] [answered]
I'm running Ubuntu with two NVIDIA GPUs. Whenever h265 codec is used in Rustdesk, nvidia-smi is indicating that rustdesk is running on GPU-0 even though I have configured X to be run exclusively on GPU-1 (GPU-0 should be reserved for my CUDA applications). It would be great if Rustdesk could be told to run exclusively on a specific GPU. 
<img width="461" alt="nvidia-smi" src="https://github.com/rustdesk/rustdesk/assets/136456291/3a825958-8292-476a-9797-773d4d5c53e5">

**Comment by 21pages** [comment]
We are rewriting it, and will take this into consideration.

**Comment by rustdesk** [comment]
Must the gpu associated with the display，othewise, useless (big latency because of data transfer).

**Reply by 21pages** [reply]
Yes, otherwise there will be memory copy  between gpu and cpu.

**Reply by chris-31337** [reply]
Thank you for the immediate discussion on this subject to both of you! I am not sure if any of your replies were meant as a question for me, so please excuse me if I am misunderstanding. My displays are indeed attached to the GPU-1. GPU-0 is solely meant for my CUDA stuff, no external display cables attached.

**Reply by 21pages** [reply]
Current version doesn't have this requirement, it transfer data between gpu and cpu, so the performance is not good enough.

**Reply by 21pages** [reply]
FFmpeg does have this option, maybe we can add this option by  env.
![1689936070300](https://github.com/rustdesk/rustdesk/assets/14891774/27f6e079-0139-4f63-82ef-bf9843a2ec13)

**Reply by 21pages** [reply, marked as answer]
https://github.com/21pages/rustdesk/releases/download/nightly/rustdesk-1.2.1-x86_64-gpu-env.deb
try add env: RUSTDESK_HWCODEC_NVENC_GPU=1.

**Reply by chris-31337** [reply]
In the nightly version you provided, I can confirm that the environment parameter does lead to rustdesk solely running on GPU1, leaving GPU0 untouched. Thank you!
![image](https://github.com/rustdesk/rustdesk/assets/136456291/ed0fd454-24ec-40a5-87ff-23be31d9e67c)

**Reply by 21pages** [reply]
merged, use tomorrow nightly
